frontend/src/pages/RegisterPage.js

```diff
@@ -2,14 +2,63 @@ import Head from '../components/Head';
 import Navlink from '../components/Navlink';
 import Footer from '../components/Footer';
 import FormTemplate from '../components/FormTemplate';
-import { 
-  Form,
-  Container,
-  Row,
-  Button
-} from 'react-bootstrap';
+import Swal from 'sweetalert2';
+import { useState, useEffect } from 'react';
+import { Form, Container, Row, Button } from 'react-bootstrap';
 
 export default function RegisterPage() {
+  const [ name, setName ] = useState('');
+  const [ username, setUserName ] = useState('');
+  const [ email, setEmail ] = useState('');
+  const [ password, setPassword ] = useState('');
+  const [ passwordConfirm, setPasswordConfirm ] = useState('');
+  const [ btnActive, setBtnActive ] = useState(false);
+
+  // set submit to active if criteria is met.
+  useEffect(() => {
+    if ((name !== "" && username !== "" && email !== "" && password !== "" && passwordConfirm !== "") && (password === passwordConfirm)) {
+      setBtnActive(true)
+    } else {
+      setBtnActive(false)
+    }
```

Comment:
you can make these conditions more readable by checking just with the variable itself since i think condition is same
```suggestion
    if ((name && username && email && password && passwordConfirm)
    && (password === passwordConfirm)) {
      setBtnActive(true)
    } else {
      setBtnActive(false)
    }
```

Reply:
I'll work on it Sir.